libvast/include/vast/system/actors.hpp

```diff
@@ -395,6 +395,13 @@ using datagram_source_actor = typed_actor_fwd<
 using node_actor = typed_actor_fwd<
   // Run an invocation in the node.
   auto(atom::run, invocation)->caf::result<caf::message>,
+  // Execute a REST endpoint on this node.
+  // This allows other nodes to access REST endpoints of this node over
+  // the CAF protocol, even if this node is not running a web server.
+  // Note that nodes connected via CAF trust each other completely,
+  // so this skips all authorization and access control mechanisms
+  // that come with HTTP(s).
+  auto(atom::proxy, http_request_description)->caf::result<std::string>,
```

Comment:
IIUC the web plugin already is a component plugin; wouldn't it make more sense to add this handler to that component so we do not pollute the interface of the node actor itself?

Reply:
The web plugin has an `authorizer` component that runs in the node, but the request dispatching itself is currently handled inside the plugin. (and I'm not sure it's a good idea to keep this, the main reason is so it can persist its state)

But we can of course introduce a new `rest_proxy` component, that's the second option listed in the PR description. It probably shouldn't depend on the `web` plugin, since all the endpoints are also native plugins. The main downside would be that it introduces one additional routing hop between the web plugin and the endpoint, so increased latency.

Reply:
I think I would prefer the second option from a code splitting perspective. Can you explain the extra hop? That doesn't quite make sense to me yet.